adr/0251-use-catalyst-for-client-side-behaviours.md

```diff
@@ -0,0 +1,28 @@
+# 251. Use Catalyst for client-side behaviours
+
+Date: 2022-04-12
+
+## Status
+
+Accepted
+
+## Context
+
+<!-- The issue motivating this decision, and any context that influences or constrains the decision. -->
+
+Existing Primer ViewComponents behaviours don't use [Catalyst](https://github.github.io/catalyst) or any other dependency, while GitHub developers often start writing a component using Catalyst.
+
+Removing the Catalyst dependency from the Web Component is an extra burden on Web Developers with few benefits. The library is ~1kb of JavaScript, and github.com already includes the library as part of its client-side bundle. Adding the library to Primer ViewComponents does not affect github.com and has a minimal effect on other applications using Primer ViewComponents.
```

Comment:
```suggestion
Removing the Catalyst dependency from the Web Component is an extra burden on Web Developers with few benefits. The library is ~1kb of JavaScript, and github.com already includes the library as part of its client-side bundle. Adding the library to primer_view_components does not affect github.com and has a minimal effect on other applications using primer_view_components.
```

Reply:
Is `primer_view_components` how we prefer to refer to the project?

Reply:
That's technically the name of the gem. Feel free to ignore this though.

Reply:
IMO we should use Primer ViewComponents or PVC. I don't think we refer to the project by the gem name anywhere in writing.